c10/util/Exception.cpp

```diff
@@ -119,6 +119,18 @@ WarningHandler* get_warning_handler() noexcept(true) {
   return ThreadWarningHandler::get_handler();
 }
 
+bool warn_only_once = true;
```

Comment:
Putting this in context.h and using the same pattern that @kurtamohler did for `_deterministic_ should be a good model: 

https://github.com/pytorch/pytorch/blob/9417e92722f1b6f75e162cbe661d3ed788d8d37a/aten/src/ATen/Context.h#L142

Reply:
Like `_determinstic` this also needs to expose the toggling of the variable to Python so we can call it from the Python test suite.

Reply:
It cannot be in `aten/src/ATEN/Context.h` since the macro is defined in `c10/util/Exception.h` and that is included in `Context.h`, so the value that is toggled must be included before the definition of the macro. I left it in `c10::Warnings`.

Reply:
Sounds good.